trunk/src/kernel/srs_kernel_mp4.cpp

```diff
@@ -5062,21 +5022,15 @@ srs_error_t SrsMp4SampleManager::write_track(SrsFrameType track,
         if (sample->frame_type == SrsVideoAvcFrameTypeKeyFrame) {
             stss_entries.push_back(sample->index + 1);
         }
-        
-        if (stts) {
-            if (previous) {
-                uint32_t delta = (uint32_t)(sample->dts - previous->dts);
-                if (stts_entry.sample_delta == 0 || stts_entry.sample_delta == delta) {
-                    stts_entry.sample_delta = delta;
-                    stts_entry.sample_count++;
-                } else {
-                    stts_entries.push_back(stts_entry);
-                    stts_entry.sample_count = 1;
-                    stts_entry.sample_delta = delta;
-                }
-            } else {
-                // The first sample always in the STTS table.
-                stts_entry.sample_count++;
+
+        if (stts && previous) {
+            if (sample->dts >= previous->dts && previous->nb_subsamples > 0) {
+                uint32_t delta = (uint32_t)(sample->dts - previous->dts) / previous->nb_subsamples;
+                stts_entry.sample_count = previous->nb_subsamples;
```

Comment:
Sample in samples is an audio/video frame, nb_subsamples is the nalu count of this frame.
In most cases, it is always one nalu in one frame, but in multi-slice frames, there is more than one nalu in one video frame.

Reply:
`nb_subsamples` is a mistake. revert this code.